### ISSUE TITLE
SwiftFormat doesn't detect .swiftformat config

### ISSUE THREAD

**Issue by mgrebenets** [nicklockwood/SwiftFormat#328]
I have created `.swiftformat` in the project root folder and I'm running `swiftformat` from the location where `.swiftformat` is placed.

However, SwiftFormat ignores `.swiftformat` and applies default rules.

```bash
$ swiftformat --lint .
swiftformat completed. 89/116 files would have been updated in 0.18s
```

If I specify location of config file, it then uses it:

```bash
$ swiftformat --lint --config .swiftformat .
swiftformat completed. 5/116 files would have been updated in 0.18s
```

Shouldn't `.swiftformat` be picked up by default?

**Comment by nicklockwood** [comment]
Yes, that should work without manually specifying the config location. I’ll see if I can reproduce.

**Comment by mgrebenets** [comment]
The path has no spaces and all characters are from Latin charset. 

One specific thing about the setup is that `swiftformat` is located in `scripts/bin/` in the root of our project repo. We did that to bundle up the tool with the repo since we can’t run brew on our ci. And we have `scripts/bin` added to the path. So when running `swiftformat` from the location where `scripts` folder is sitting it pocks up the bundled executable. 

Probably that’s what confuses the logic for picking up a config. 

We also have some `Scripts` and `scripts` naming mixed up. Git doesn’t differentiate based on the case, but maybe logic in swiftformat is case-sensitive?

**Comment by nicklockwood** [comment]
> Probably that’s what confuses the logic for picking up a config.

I can't quite visualize what you meant, but if it helps, SwiftFormat can only detect config files inside folders that you ask it to format, not in ones above that level in the tree. So if your project structure is something like:

```
Project
    .swiftformat
    Source/
        Class.swift
```

And you run `$ swiftformat Source`, that won't work because swiftformat will only scan the Source folder and will never look at the Project root.

As for the "Scripts" vs "scripts" issue, then yes, it's possible that this would screw up both the glob matching and cache performance, which is based on (case-sensitive) string matching.
